File: Php/app/Cashgame/CashgameValidatorFactory.php
<?php
namespace app\Cashgame{

	use app\Cashgame\Action\ActionPostModel;
	use app\Cashgame\Action\BuyinPostModel;
	use core\Validation\Validator;
	use entities\Cashgame;
	use app\Cashgame\Edit\CashgameEditPostModel;
	use entities\Homegame;

	interface CashgameValidatorFactory{

		/**
		 * @param Homegame $homegame
		 * @param Cashgame $cashgame
		 * @return Validator
		 */
		public function getAddCashgameValidator(Homegame $homegame, Cashgame $cashgame);

		/**
		 * @param CashgameEditPostModel $postModel
		 * @return Validator
		 */
		public function getEditCashgameValidator(CashgameEditPostModel $postModel);

		/**
		 * @param BuyinPostModel $postModel
		 * @return Validator
		 */
		public function getBuyinValidator(BuyinPostModel $postModel);

		/**
		 * @param ActionPostModel $postModel
		 * @return Validator
		 */
		public function getCashoutValidator(ActionPostModel $postModel);

		/**
		 * @param ActionPostModel $postModel
		 * @return Validator
		 */
		public function getReportValidator(ActionPostModel $postModel);

	}

}

File: Web/Validators/CashgameValidatorFactory.cs
using Core.Classes;

namespace Web.Validators{

	public class CashgameValidatorFactory : ICashgameValidatorFactory{

		public Validator GetAddCashgameValidator(Homegame homegame, Cashgame cashgame){
			var validator = new CompositeValidator();
			validator.AddValidator(new RequiredValidator(cashgame.Location, "Location can't be empty"));
			return validator;
		}

        /*
		public function getEditCashgameValidator(CashgameEditPostModel $postModel){
			$validator = new CompositeValidator();
			$validator.addValidator(new RequiredValidator($postModel.location, 'Location can\'t be empty'));
			return $validator;
		}

		public function getBuyinValidator(BuyinPostModel $postModel){
			$validator = new CompositeValidator();
			$validator.addValidator(new RequiredValidator($postModel.amount, "Amount can't be empty"));
			$validator.addValidator(new PositiveNumberValidator($postModel.amount, 'Amount needs to be positive'));
			$validator.addValidator(new PositiveNumberValidator($postModel.stack, "Stack size can't be negative"));
			return $validator;
		}

		public function getCashoutValidator(ActionPostModel $postModel){
			$validator = new CompositeValidator();
			$validator.addValidator(new RequiredValidator($postModel.stack, "Stack can't be empty"));
			$validator.addValidator(new PositiveNumberValidator($postModel.stack, "Stack size can't be negative"));
			return $validator;
		}

		public function getReportValidator(ActionPostModel $postModel){
			$validator = new CompositeValidator();
			$validator.addValidator(new RequiredValidator($postModel.stack, "Stack can't be empty"));
			$validator.addValidator(new PositiveNumberValidator($postModel.stack, "Stack size can't be negative"));
			return $validator;
		}
        */

	}

}